python/lsst/ip/isr/crosstalk.py

```diff
@@ -0,0 +1,205 @@
+#
+# LSST Data Management System
+# Copyright 2008-2017 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+"""
+Apply intra-CCD crosstalk corrections
+"""
+from __future__ import absolute_import, division, print_function
+
+__all__ = ["CrosstalkConfig", "CrosstalkTask", "subtractCrosstalk"]
+
+from builtins import zip
+
+import numpy as np
+
+import lsst.afw.math
+import lsst.afw.table
+import lsst.afw.detection
+
+from lsst.pex.config import Config, Field, ListField
+from lsst.pipe.base import Task
+
+import lsst.afw.display
+display = lsst.afw.display.Display(backend="ds9", frame=1)
+
+
+class CrosstalkConfig(Config):
+    """Configuration for intra-CCD crosstalk removal"""
+    minPixelToMask = Field(dtype=float, default=45000,
+                           doc="Set crosstalk mask plane for pixels over this value")
+    crosstalkMaskPlane = Field(dtype=str, default="CROSSTALK", doc="Name for crosstalk mask plane")
+    coeffs = ListField(
+        dtype=float,
+        doc=("Flattened crosstalk coefficient matrix; should have nAmps x nAmps entries. "
+             "Once 'reshape'-ed, ``coeffs[i][j]`` is the fraction of the j-th amp present on the i-th amp."),
+        default=[0, 0, 0, 0,
+                 0, 0, 0, 0,
+                 0, 0, 0, 0,
+                 0, 0, 0, 0],
+    )
+
+    def getCoeffs(self, numAmps):
+        """Return a 2-D numpy array of crosstalk coefficients of the proper shape"""
+        return np.array(self.coeffs).reshape((numAmps, numAmps))
+
+
+class CrosstalkTask(Task):
+    """Apply intra-CCD crosstalk correction"""
+    ConfigClass = CrosstalkConfig
+
+    def run(self, exposure):
```

Comment:
I think that since the coefficients can't be in the config, this signature needs to take either a `dataRef` or an opaque object that contains the details of the crosstalk.